File: Mudit Choraria/count-sorted-vowel-strings.cpp
class Solution {
public:
    int countVowels(int id, int n, vector<vector<int>>& dp) {
        if(n == 0) {
            return 1;
        }
        if(id == 0) {
            return 0;
        }
        if(dp[id][n] != -1) {
            return dp[id][n];
        }
        return dp[id][n] = countVowels(id - 1, n, dp) + countVowels(id, n - 1, dp);
    }
    int countVowelStrings(int n) {
        vector<vector<int>> dp(6, vector<int>(n + 1, -1));
        return countVowels(5, n, dp);
    }
};

File: Harish-Kumar/Count Sorted Vowel Strings.java
/* 

https://leetcode.com/problems/count-sorted-vowel-strings/
*/

class Solution {
    public int countVowelStrings(int n) {
        int a[] = { 1, 1, 1, 1, 1 };
        for (int i = 2; i <= n; i++) {
            a[0] = a[0] + a[1] + a[2] + a[3] + a[4];
            a[1] = a[1] + a[2] + a[3] + a[4];
            a[2] = a[2] + a[3] + a[4];
            a[3] = a[3] + a[4];
        }
        return a[0] + a[1] + a[2] + a[3] + a[4];
    }
}